docs/src/content/docs/core/routing.mdx

```diff
@@ -111,6 +115,66 @@ Return values:
 
 ---
 
+## HTTP Method Routing
+
+You can handle different HTTP methods (GET, POST, PUT, DELETE, etc.) on the same path by passing an object with method keys:
+
+```tsx
+route("/api/users", {
+  get: () => new Response(JSON.stringify(users)),
+  post: ({ request }) => new Response("User created", { status: 201 }),
+  delete: () => new Response("User deleted", { status: 204 }),
+});
+```
```

Comment:
@peterp how does this API look to you? It is also what I had in mind for when we supported methods, but also want to see what you think / how it maps to what you might've imagined for this.